packages/mui-material/src/ButtonBase/ButtonBase.js

```diff
@@ -303,15 +297,15 @@ const ButtonBase = React.forwardRef(function ButtonBase(inProps, ref) {
   if (process.env.NODE_ENV !== 'production') {
     // eslint-disable-next-line react-hooks/rules-of-hooks
     React.useEffect(() => {
-      if (enableTouchRipple && !rippleRef.current) {
+      if (enableTouchRipple && !ripple.ref) {
         console.error(
           [
             'MUI: The `component` prop provided to ButtonBase is invalid.',
             'Please make sure the children prop is rendered in this custom component.',
           ].join('\n'),
         );
       }
-    }, [enableTouchRipple]);
+    }, [enableTouchRipple, ripple.ref]);
```

Comment:
Because the ripple is now lazy, checking for the presence of the ripple to determine if `ButtonBase`'s `component` renders `children` is now not good/optimal. I could remove this dev warning or replace it with something else if someone has an idea for a good test for this.